src/main/java/io/tackle/pathfinder/services/AssessmentSvc.java

```diff
@@ -341,11 +343,12 @@ private AssessmentQuestionnaire copyQuestionnaireBetweenAssessments(Assessment s
 
     @Transactional
     public List<LandscapeDto> landscape(List<Long> applicationIds) {
-            String sql = "SELECT cast(ID as int), RISK, cast(trunc(max(PCT)) as int) AS PERCENTAGE " +
+            String sql = "SELECT cast(ID as int), RISK, cast(trunc(max(PCT)) as int) AS PERCENTAGE , cast(application_id as int) " +
```

Comment:
@jonathanvila  `cast(application_id as int)` made me think what would happen if the `applicationI`d is greater than the max value of an integer; the `applicationId` is a value coming from `tackle-application-inventory` and is defined as `Long`. 

I did local tests to verify that numbers big enough to not fit into the `Integer` definition but only on `Long` cause exceptions. So `cast(application_id as int)` will cause problems. In order to reproduce the error you only need to:
- Change this line https://github.com/konveyor/tackle-pathfinder/blob/main/src/test/java/io/tackle/pathfinder/controllers/AssessmentsResourceTest.java#L759 to `.body(new ApplicationDto(Long.MAX_VALUE))`
- Change this line https://github.com/konveyor/tackle-pathfinder/blob/main/src/test/java/io/tackle/pathfinder/controllers/AssessmentsResourceTest.java#L816 to `.body(List.of(new ApplicationDto(659500L), new ApplicationDto(Long.MAX_VALUE)))`

Is there a way we can avoid casting to `int` so `applicationId (LONG)` can be fully supported?
I'm aware that this is not the only place where we use `cast` so if we find a solution for this one we might also apply the same logic in other places (if applicable)

Reply:
:+1:  I will check.